test/test_unit_keystore_credentials_provider.py

```diff
@@ -28,18 +38,68 @@ def test_save_can_raise_operation_not_complete_exception(self, mock_store_adapte
+        self.assertEquals(MockCredentials.login, credentials_data.login)
+        self.assertEquals(MockCredentials.password, credentials_data.password)
 
     @patch.object(KeystoreAdapter, 'set_password')
     def test_update_api_key_calls_methods_properly(self, mock_store_adapter):
         # mock_store_adapter.set_password.return_value=None
         credential_provider = KeystoreCredentialsProvider()
         credential_provider.update_api_key_entry('someusertoupdate', MockCredentials, 'newapikey')
-        calls = [call('https://somemachine', 'machine', 'https://somemachine'), call('https://somemachine', 'login', 'someusertoupdate'), call('https://somemachine', 'password', 'newapikey') ]
+        calls = [call(TEST_HOSTNAME, MACHINE, TEST_HOSTNAME),
+                 call(TEST_HOSTNAME, LOGIN, 'someusertoupdate'),
+                 call(TEST_HOSTNAME, PASSWORD, 'newapikey')]
         mock_store_adapter.assert_has_calls(calls)
 
     @patch.object(KeystoreAdapter, 'set_password', side_effect=Exception)
     def test_update_api_key_can_raise_operation_not_complete_exception(self, mock_store_adapter):
         with self.assertRaises(OperationNotCompletedException):
             credential_provider = KeystoreCredentialsProvider()
             credential_provider.update_api_key_entry('someusertoupdate', MockCredentials, 'newapikey')
+
+    @patch.object(KeystoreAdapter, "get_password", return_value=None)
+    def test_is_exists_return_false_if_attr_not_exists(self, mock_store_adapter):
+        credential_provider = KeystoreCredentialsProvider()
+        self.assertEquals(False, credential_provider.is_exists(TEST_HOSTNAME))
+
+    @patch.object(KeystoreAdapter, "get_password", return_value="password")
+    def test_is_exists_return_true_when_attr_exists(self, mock_store_adapter):
+        credential_provider = KeystoreCredentialsProvider()
+        self.assertEquals(True, credential_provider.is_exists(TEST_HOSTNAME))
+
+    @patch.object(KeystoreAdapter, "get_password", side_effect=Exception)
+    def test_is_exists_raises_exception_when_exception_is_thrown_by_get_password(self, mock_store_adapter):
+        credential_provider = KeystoreCredentialsProvider()
+        with self.assertRaises(Exception):
+            credential_provider.is_exists(TEST_HOSTNAME)
+
+    @patch.object(KeystoreAdapter, 'delete_password', return_value=None)
+    @patch.object(KeystoreAdapter, "get_keyring_name", return_value=TEST_KEYRING)
+    def test_remove_credentials_calls_delete_password(self, mock_keystore_adapter, another_mock_keystore_adatper):
+        credential_provider = KeystoreCredentialsProvider()
+        credential_provider.remove_credentials(MockConjurrcData)
+        another_mock_keystore_adatper.assert_called_once_with(TEST_HOSTNAME)
+
+    @patch.object(KeystoreAdapter, 'delete_password', side_effect=keyring.errors.KeyringError)
+    @patch.object(KeystoreAdapter, "get_keyring_name", return_value=TEST_KEYRING)
```

Comment:
Sorry I didnt see this sooner but @tzheleznyak NIT some function names are wrapped in `''` and some `""`